std/container/array.d

```diff
@@ -349,7 +356,7 @@ if (!is(Unqual!T == bool))
             import core.checkedint : mulu;
             bool overflow;
             const sz = mulu(elements, T.sizeof, overflow);
-            if (overflow) assert(0);
+            assert(!overflow, "Trying to allocate more than size_t.max bytes.");
```

Comment:
Does it? I thought all asserts are removed in release mode, so it shouldn't change anything. Am I wrong?